In [1]:
#imports
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

#sklearn stuff
from sklearn import preprocessing
from sklearn.preprocessing import MinMaxScaler, StandardScaler
from sklearn.model_selection import train_test_split, cross_val_score, GridSearchCV
from sklearn.metrics import mean_squared_error, r2_score
from sklearn.ensemble import RandomForestRegressor, GradientBoostingRegressor
from sklearn.svm import SVR
from sklearn.neural_network import MLPRegressor
from sklearn.decomposition import PCA

from math import sqrt
%matplotlib inline

In [2]:
# Import the Dataset
data = pd.read_csv('Capstone_Dataset.csv')
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 365 entries, 0 to 364
Data columns (total 11 columns):
 #   Column                                                                            Non-Null Count  Dtype  
---  ------                                                                            --------------  -----  
 0   Date                                                                              365 non-null    object 
 1   WTI Price (Dollars Per Barrel)                                                    365 non-null    float64
 2   U.S. Imports of Crude Oil (Thousand Barrels)                                      365 non-null    int64  
 3   U.S. Exports of Crude Oil (Thousand Barrels)                                      365 non-null    int64  
 4   U.S. Crude Oil Rotary Rigs
in Operation (Count)                                   365 non-null    int64  
 5   U.S. Field Production of Crude Oil (Thousand Barrels per Day)                     365 non-null    int64  
 6   Ea

In [3]:
data.head(10)

Date  WTI Price (Dollars Per Barrel)  \
0  Jan-1990                           22.86   
1  Feb-1990                           22.11   
2  Mar-1990                           20.39   
3  Apr-1990                           18.43   
4  May-1990                           18.20   
5  Jun-1990                           16.70   
6  Jul-1990                           18.45   
7  Aug-1990                           27.31   
8  Sep-1990                           33.51   
9  Oct-1990                           36.04   

   U.S. Imports of Crude Oil (Thousand Barrels)  \
0                                        192572   
1                                        165046   
2                                        189625   
3                                        174393   
4                                        200067   
5                                        192704   
6                                        212514   
7                                        200000   
8                                        169907   
9                                        159080   

   U.S. Exports of Crude Oil (Thousand Barrels)  \
0                                          4084   
1                                          2843   
2                                          4104   
3                                          3338   
4                                          3474   
5                                          2639   
6                                          2762   
7                                          1990   
8                                          2050   
9                                          3233   

   U.S. Crude Oil Rotary Rigs\nin Operation (Count)  \
0                                               514   
1                                               492   
2                                               478   
3                                               484   
4                                               493   
5                                               498   
6                                               497   
7                                               506   
8                                               566   
9                                               588   

   U.S. Field Production of Crude Oil (Thousand Barrels per Day)  \
0                                               7546               
1                                               7497               
2                                               7433               
3                                               7407               
4                                               7328               
5                                               7106               
6                                               7173               
7                                               7287               
8                                               7224               
9                                               7542               

   East Coast (PADD 1) Field Production of Crude Oil (Thousand Barrels per Day)  \
0                                                 35                              
1                                                 35                              
2                                                 31                              
3                                                 32                              
4                                                 31                              
5                                                 29                              
6                                                 27                              
7                                                 28                              
8                                                 27                              
9                                                 28                              

   Midwest (PADD 2) Field Production of Crude Oil (Thousand Barrels per Day)  \
0               

In [5]:
# Rename columns
data['Oil_Price'] = data['WTI Price (Dollars Per Barrel)']
data['Oil_Imports'] = data['U.S. Imports of Crude Oil (Thousand Barrels)']
data['Oil_Exports'] = data['U.S. Exports of Crude Oil (Thousand Barrels)']
data['Rig_Count'] = data['U.S. Crude Oil Rotary Rigs\nin Operation (Count)']
data['U.S._Oil_Production'] = data['U.S. Field Production of Crude Oil (Thousand Barrels per Day)']
data['EastCoast_Oil_Production'] = data['East Coast (PADD 1) Field Production of Crude Oil (Thousand Barrels per Day)']
data['Midwest_Oil_Production'] = data['Midwest (PADD 2) Field Production of Crude Oil (Thousand Barrels per Day)']
data['GulfCoast_Oil_Production'] = data['Gulf Coast (PADD 3) Field Production of Crude Oil (Thousand Barrels per Day)']
data['RockyMountain_Oil_Production'] = data['Rocky Mountain (PADD 4) Field Production of Crude Oil (Thousand Barrels per Day)']
data['WestCoast_Oil_Production'] = data['West Coast (PADD 5) Field Production of Crude Oil (Thousand Barrels per Day)']

# Drop unneeded columns
data = data.drop(['Date', 'WTI Price (Dollars Per Barrel)', 'U.S. Imports of Crude Oil (Thousand Barrels)',
                 'U.S. Exports of Crude Oil (Thousand Barrels)', 'U.S. Crude Oil Rotary Rigs\nin Operation (Count)',
                 'U.S. Field Production of Crude Oil (Thousand Barrels per Day)',
                 'East Coast (PADD 1) Field Production of Crude Oil (Thousand Barrels per Day)',
                 'Midwest (PADD 2) Field Production of Crude Oil (Thousand Barrels per Day)',
                 'Gulf Coast (PADD 3) Field Production of Crude Oil (Thousand Barrels per Day)',
                 'Rocky Mountain (PADD 4) Field Production of Crude Oil (Thousand Barrels per Day)',
                 'West Coast (PADD 5) Field Production of Crude Oil (Thousand Barrels per Day)'], axis=1)

data.head()

Oil_Price  Oil_Imports  Oil_Exports  Rig_Count  U.S._Oil_Production  \
0      22.86       192572         4084        514                 7546   
1      22.11       165046         2843        492                 7497   
2      20.39       189625         4104        478                 7433   
3      18.43       174393         3338        484                 7407   
4      18.20       200067         3474        493                 7328   

   EastCoast_Oil_Production  Midwest_Oil_Production  GulfCoast_Oil_Production  \
0                        35                     785                      3374   
1                        35                     744                      3401   
2                        31                     745                      3348   
3                        32                     736                      3354   
4                        31                     737                      3310   

   RockyMountain_Oil_Production  WestCoast_Oil_Production  
0                           505                      2847  
1                           508                      2809  
2                           508                      2801  
3                           502                      2783  
4                           506                      2744

In [6]:
# Check the data types
data.dtypes

Oil_Price                       float64
Oil_Imports                       int64
Oil_Exports                       int64
Rig_Count                         int64
U.S._Oil_Production               int64
EastCoast_Oil_Production          int64
Midwest_Oil_Production            int64
GulfCoast_Oil_Production          int64
RockyMountain_Oil_Production      int64
WestCoast_Oil_Production          int64
dtype: object

In [7]:
# Drop duplicate rows
data = data.drop_duplicates()

# Check for nulls
nulls = data.isnull().sum()
nulls

Oil_Price                       0
Oil_Imports                     0
Oil_Exports                     0
Rig_Count                       0
U.S._Oil_Production             0
EastCoast_Oil_Production        0
Midwest_Oil_Production          0
GulfCoast_Oil_Production        0
RockyMountain_Oil_Production    0
WestCoast_Oil_Production        0
dtype: int64

In [8]:
# View the makeup of the data
data.describe()

Oil_Price    Oil_Imports    Oil_Exports    Rig_Count  \
count  365.000000     365.000000     365.000000   365.000000   
mean    47.586082  250595.868493   10582.882192   503.287671   
std     29.108127   42389.848480   22408.439895   378.852543   
min     11.350000  142930.000000      52.000000   108.000000   
25%     21.280000  221130.000000     975.000000   244.000000   
50%     40.750000  248909.000000    2241.000000   358.000000   
75%     67.870000  284528.000000    4646.000000   668.000000   
max    133.880000  327476.000000  113731.000000  1596.000000   

       U.S._Oil_Production  EastCoast_Oil_Production  Midwest_Oil_Production  \
count           365.000000                365.000000              365.000000   
mean           6938.547945                 30.693151              888.528767   
std            1885.005407                 14.760695              567.528309   
min            3974.000000                 14.000000              412.000000   
25%            5607.000000                 21.000000              473.000000   
50%            6442.000000                 26.000000              596.000000   
75%            7433.000000                 31.000000             1248.000000   
max           12860.000000                 83.000000             2332.000000   

       GulfCoast_Oil_Production  RockyMountain_Oil_Production  \
count                365.000000                    365.000000   
mean                3879.709589                    463.471233   
std                 1417.490226                    185.486167   
min                 1664.000000                    281.000000   
25%                 3135.000000                    349.000000   
50%                 3255.000000                    383.000000   
75%                 4047.000000                    502.000000   
max                 8660.000000                   1021.000000   

       WestCoast_Oil_Production  
count                365.000000  
mean                1676.153425  
std                  601.029107  
min                  803.000000  
25%                 1134.000000  
50%                 1597.000000  
75%                 2212.000000  
max                 2878.000000

The price of oil ranges from 11.35 to 133.88 with a mean of 47.58 and standard deviation of 29.1. The Gulf Coast Region has the highest production of all the Regions with a mean of 3879.7 Thousand Barrels Per Day and the East Coast Region has the lowest production with a mean of 30.69 Thousand Barrels Per Day.

In [15]:
# Check number of distinct values for Oil Price
data['Oil_Price'].value_counts().shape

(348,)

In [9]:
# Check the correlations of the variables
data.corr()

Oil_Price  Oil_Imports  Oil_Exports  Rig_Count  \
Oil_Price                      1.000000     0.300876     0.081131   0.645806   
Oil_Imports                    0.300876     1.000000    -0.325868  -0.290850   
Oil_Exports                    0.081131    -0.325868     1.000000   0.259742   
Rig_Count                      0.645806    -0.290850     0.259742   1.000000   
U.S._Oil_Production            0.019949    -0.590451     0.867794   0.465431   
EastCoast_Oil_Production       0.138084    -0.494174     0.811246   0.533553   
Midwest_Oil_Production         0.337257    -0.408121     0.770525   0.667703   
GulfCoast_Oil_Production       0.169090    -0.390281     0.911947   0.474204   
RockyMountain_Oil_Production   0.223246    -0.498985     0.849272   0.544584   
WestCoast_Oil_Production      -0.726984    -0.379709    -0.438754  -0.470431   

                              U.S._Oil_Production  EastCoast_Oil_Production  \
Oil_Price                                0.019949                  0.138084   
Oil_Imports                             -0.590451                 -0.494174   
Oil_Exports                              0.867794                  0.811246   
Rig_Count                                0.465431                  0.533553   
U.S._Oil_Production                      1.000000                  0.953711   
EastCoast_Oil_Production                 0.953711                  1.000000   
Midwest_Oil_Production                   0.920202                  0.939823   
GulfCoast_Oil_Production                 0.963175                  0.939674   
RockyMountain_Oil_Production             0.960199                  0.946852   
WestCoast_Oil_Production                -0.324072                 -0.429356   

                              Midwest_Oil_Production  \
Oil_Price                                   0.337257   
Oil_Imports                                -0.408121   
Oil_Exports                                 0.770525   
Rig_Count                                   0.667703   
U.S._Oil_Production                         0.920202   
EastCoast_Oil_Production                    0.939823   
Midwest_Oil_Production                      1.000000   
GulfCoast_Oil_Production                    0.944143   
RockyMountain_Oil_Production                0.962669   
WestCoast_Oil_Production                   -0.605206   

                              GulfCoast_Oil_Production  \
Oil_Price                                     0.169090   
Oil_Imports                                  -0.390281   
Oil_Exports                                   0.911947   
Rig_Count                                     0.474204   
U.S._Oil_Production                           0.963175   
EastCoast_Oil_Production                      0.939674   
Midwest_Oil_Production                        0.944143   
GulfCoast_Oil_Production                      1.000000   
RockyMountain_Oil_Production                  0.951945   
WestCoast_Oil_Production                     -0.546096   

                              RockyMountain_Oil_Production  \
Oil_Price                                         0.223246   
Oil_Imports                                      -0.498985   
Oil_Exports                                       0.849272   
Rig_Count                                         0.544584   
U.S._Oil_Production                               0.960199   
EastCoast_Oil_Production                          0.946852   
Midwest_Oil_Production                            0.962669   
GulfCoast_Oil_Production                          0.951945   
RockyMountain_Oil_Production                      1.000000   
WestCoast_Oil_Production                         -0.474619   

                              WestCoast_Oil_Production  
Oil_Price                                    -0.726984  
Oil_Imports                                  -0.379709  
Oil_Exports                                  -0.438754  
Rig_Count                                    -0.470431  
U.S._Oil_Production                          -0.324072 

The variables West Coast Oil Production and Rig Count have the highest correlation with Oil Price in the dataset. U.S. Oil Production and Oil Exports have the lowest correlation with Oil Price in the dataset.

All datatypes are numeric values, there are no duplicate rows or null values, the column 'Date' has been removed and all column names have been changed.

In [10]:
# Save the cleaned dataset in a csv file
data.to_csv('Capstone_Dataset_Cleaned.csv', index=False)